client_test.go

```diff
@@ -62,3 +62,19 @@ func TestClientKnownGoodEndpoint(t *testing.T) {
 		t.Errorf("Endpoint is not https://sentry.io/api/0 got %s", bclient.Endpoint)
 	}
 }
+
+func TestNewRequestWillNotAddExtraTrailingSlashToEndpoint(t *testing.T) {
+	endpoint := "some-endpoint/"
+	bclient, berr := NewClient("testauthclient", nil, nil)
+	if berr != nil {
+		t.Error(berr)
+	}
+	req, err := bclient.newRequest("get", endpoint, nil)
+	if req == nil || err != nil {
+		t.Errorf("can't genearte request: %v", err)
```

Comment:
```suggestion
		t.Errorf("can't generate request: %v", err)
```

Reply:
Not sure if we want to separate this error handling since if req == nil and err == nil, it will just append a nil err